xpra/server/window/window_source.py

```diff
@@ -1082,6 +1088,7 @@ def is_cancelled(self, sequence=INFINITY):
 
 
     def calculate_batch_delay(self, has_focus, other_is_fullscreen, other_is_maximized):
+        self.has_focus = has_focus
```

Comment:
As I mentioned earlier, I don't like having a flag that's updated on a timer when we have better data available.
Also:
* why is focus so important?
* why not take into account OR windows?

We could use `window.get("focused", False)` every time `damage` is entered.

Reply:
The intent of focus here is to try to ensure we have nvenc initialized and used for windows that are in focus/active regardless of what type of window they are to keep responsiveness high while letting other windows use their previous scoring (ie text windows not using video encoders at all or video subregions holding on to a video encoder during the entire time video is being used).
I'm not very familiar with all aspects of the code, so if there's a better way of getting this info that is performant enough (using window get focused or otherwise) I'm sure that'll work better. This was just where `has_focus` was available in window_source to capture. In this scenario, `has_focus` is only needed as a best effort, so if its not precisely accurate it doesn't hurt anything - but as you mentioned, that could change in the future or other things could use that param that relies on it being accurate.
I'll try out `self.has_focus = window.get("focused", False)` in `damage` and see if that works as expected